framework/include/utils/InputParameterWarehouse.h

```diff
@@ -23,7 +23,32 @@
 class InputParameters;
 class MooseApp;
 
-typedef std::map<std::string, MooseSharedPointer<InputParameters> >::iterator InputParameterIterator;
+/**
+ * A storage container for the name of parameters to be controlled
+ */
+struct InputParametersContainer
+{
+  /// The "system" name (e.g., Kernels)
+  std::string system;
```

Comment:
This container exists for one reason, accessing the objects with controls given the various possible names. You might be right that system, syntax, and tag could be combined, I will take a look.